ingest/sqs/system/delivery/http/system_http_handler.go

```diff
@@ -1,17 +1,130 @@
 package http
 
 import (
+	"encoding/json"
+	"fmt"
+	"io"
 	"net/http"
 	_ "net/http/pprof"
+	"os"
+
+	"go.uber.org/zap"
+
+	"github.com/go-redis/redis"
+
+	"github.com/osmosis-labs/osmosis/v20/ingest/sqs/domain/mvc"
+	"github.com/osmosis-labs/osmosis/v20/ingest/sqs/log"
 
 	"github.com/labstack/echo"
 )
 
+type SystemHandler struct {
+	logger    log.Logger
+	host      string // Host for the GRPC gateway
+	grpcPort  string // GRPC gateway port
+	redisHost string // Redis host
+	redisPort string // Redis port
+	CIUsecase mvc.ChainInfoUsecase
+}
+
 // NewSystemHandler will initialize the /debug/ppof resources endpoint
-func NewSystemHandler(e *echo.Echo) {
+func NewSystemHandler(e *echo.Echo, logger log.Logger, us mvc.ChainInfoUsecase) {
+	// GRPC Gateway configuration
+	host := getEnvOrDefault("GRPC_GATEWAY_HOST", "localhost")
+	grpcPort := getEnvOrDefault("GRPC_GATEWAY_PORT", "1317")
+
+	// Redis configuration
+	redisHost := getEnvOrDefault("REDIS_HOST", "localhost")
+	redisPort := getEnvOrDefault("REDIS_PORT", "6379")
```

Comment:
Let's set all these in one place in `app.go`. We will later move all configs into a file instead of keeping that in environment variables.